Ghidra/Processors/CP1600/data/languages/CP1600.slaspec

```diff
@@ -0,0 +1,651 @@
+	local ptr:2 = addr16;
+	comparison(reg0_2, *(*:2 ptr));
+}
+
+:CMPR reg3_5, reg0_2 is opcode6_9=0x0005 & reg3_5 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	comparison(reg0_2, reg3_5);
+}
+
+:COMR reg0_2 is opcode3_9=0x0003 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	reg0_2 = ~reg0_2;
+	resultFlags(reg0_2);
+}
+
+:DECR reg0_2 is opcode3_9=0x0002 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	reg0_2 = reg0_2 - 1;
+	resultFlags(reg0_2);
+}
+
+:DIS is opcode0_9=0x0003 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	I = 0;
+}
+
+:EIS is opcode0_9=0x0002 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	I = 1;
+}
+
+:GSWD reg0_1 is opcode2_9=0x000C & reg0_1 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local mask:2 = (zext(S) << 7) + (zext(Z) << 6) + (zext(O) << 5) + (zext(C) << 4);
+	reg0_1 = (mask << 8) + mask; 
+}
+
+:HLT is opcode0_9=0x0000 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	goto inst_start;
+}
+
+:INCR reg0_2 is opcode3_9=0x0001 & reg0_2 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	reg0_2 = reg0_2 + 1;
+	resultFlags(reg0_2);
+}
+
+:J jmpdest16 is opcode0_9=0x0004 ; jump_type=0 & reg24_25=R7 & jmpdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	goto jmpdest16;
+}
+
+:JD jmpdest16 is opcode0_9=0x0004 ; jump_type=2 & reg24_25=R7 & jmpdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	I = 0;
+	goto jmpdest16;
+}
+
+:JE jmpdest16 is opcode0_9=0x0004 ; jump_type=1 & reg24_25=R7 & jmpdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	I = 1;
+	goto jmpdest16;
+}
+
+:JSR reg24_25, jmpdest16 is opcode0_9=0x0004 ; jump_type=0 & reg24_25 & jmpdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	reg24_25 = inst_next;
+	call jmpdest16;
+}
+
+:JSRD reg24_25, jmpdest16 is opcode0_9=0x0004 ; jump_type=2 & reg24_25 & jmpdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	I = 0;
+	reg24_25 = inst_next;
+	call jmpdest16;
+}
+
+:JSRE reg24_25, jmpdest16 is opcode0_9=0x0004 ; jump_type=1 & reg24_25 & jmpdest16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	I = 1;
+	reg24_25 = inst_next;
+	call jmpdest16;
+}
+
+:MOVR reg3_5, reg0_2 is opcode6_9=0x0002 & reg0_2 & reg3_5 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	reg0_2 = reg3_5;
+	resultFlags(reg0_2);
+}
+
+:MVI addr16, reg0_2 is opcode3_9=0x0050 & reg0_2 ; addr16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local ptr:2 = addr16;
+	reg0_2 = *(*:2 ptr);
+}
+
+:MVO reg0_2, addr16 is opcode3_9=0x0048 & reg0_2 ; addr16 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local ptr:2 = addr16;
+	*ptr = reg0_2;
+}
+
+:MVO@ reg0_2, reg3_5 is opcode6_9=0x0009 & reg0_2 & reg3_5 [ doublebyte=0; globalset(inst_next, doublebyte); ] {
+	local ptr:2 = reg3_5; 
+	*ptr = reg0_2;
+	if ((reg3_5 != R4) && (reg3_5 != R5) && (reg3_5 != R6)) goto inst_next;
```

Comment:
this syntax does not work like you'd think.  this is comparing the contents of the reg3_5 to the contents of R4, R5, and R6.  To check for R4, R5, or R6 you would need another token (`op3_5=(3,5)`)to compare against 0x4, 0x5, and 0x6

Reply:
Fixed, thanks.